### PR TITLE
Reader: update to use org id to determine P2 sites

### PR DESCRIPTION
**Pull request by ScoutHarris**
Ref: https://github.com/wordpress-mobile/WordPress-iOS/issues/15343, https://github.com/wordpress-mobile/WordPress-iOS/issues/15264
WPKit PR: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/313

This updates `ReaderPost` and `ReaderSiteTopic` to use a site's `organization_id` to determine if it is a P2 or not.

This resolves two issues:
- [P2 stream header](https://github.com/wordpress-mobile/WordPress-iOS/pull/15442): a Followed P2 will now show the P2 header.
- [Post detail](https://github.com/wordpress-mobile/WordPress-iOS/pull/15447#issuecomment-740269725): a Followed P2 will now show the P2 default icon.

To test:

---
P2 stream:
- Log in with an account that is only following a P2.
- Go to Reader > P2.
- Verify the post header is the [P2 format](https://github.com/wordpress-mobile/WordPress-iOS/pull/15315) (i.e. displays the author information).

<kbd>p2_card_header</kbd>

---
P2 post detail:
- Log in with an account that is only following a P2 that does not have a site icon set.
- Go to Reader > P2.
- Tap on the P2 card header.
- Verify the site icon is the P2 default.

<kbd>p2_default_icon</kbd>

---
PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/Models/ReaderPost.m [hunk 1]
```diff
@@ -235,6 +235,12 @@ - (BOOL)isPrivate
     return self.isBlogPrivate;
 }
 
+- (BOOL)isP2Type
```
Comment (aerych) [inline WordPress/Classes/Models/ReaderPost.m:238]: Nice

File: WordPress/Classes/ViewRelated/Reader/ReaderSiteStreamHeader.swift [hunk 2]
```diff
@@ -91,13 +91,11 @@ fileprivate func > <T: Comparable>(lhs: T?, rhs: T?) -> Bool {
         }
     }
 
-    private func configureHeaderImage(_ siteBlavatar: String?, isWPForTeams: Bool = false) {
+    private func configureHeaderImage(_ siteTopic: ReaderSiteTopic) {
```
Comment (aerych) [inline WordPress/Classes/ViewRelated/Reader/ReaderSiteStreamHeader.swift:94]: Yay for simplifying :)

File: WordPress/Classes/Models/ReaderSiteTopic.swift [hunk 3]
```diff
@@ -49,7 +54,7 @@ import Foundation
         isJetpack = remoteInfo.isJetpack
         isPrivate = remoteInfo.isPrivate
         isVisible = remoteInfo.isVisible
-        isWPForTeams = remoteInfo.isWPForTeams
+        organizationID = remoteInfo.organizationID
```
Comment (aerych) [inline WordPress/Classes/Models/ReaderSiteTopic.swift:57]: If I navigate to the discover feed (and pull to refresh if needed) organization_id is nil and an exception is thrown . Maybe `remoteInfo.organizationID ?? 0` to handle that?

**Review by aerych — CHANGES_REQUESTED** [review]
Heya @ScoutHarris 👋 
Looking pretty good.  Just found one little thing while testing. 

Back to you!

File: WordPress/Classes/Models/ReaderSiteTopic.swift (hunk 3 above)
Comment (aerych) [reply WordPress/Classes/Models/ReaderSiteTopic.swift:57]: Not quite sure why the nil is happening there.  I didn't dig into it

**Comment by ScoutHarris** [comment]
Hey @aerych . Sorry about that. Should be fixed now. Ready for another review please.

**Review by aerych — APPROVED** [review]
Yay! :shipit: !

**Pull request merged**